doc/man7/provider.pod

```diff
@@ -276,13 +323,24 @@ It currently offers the following FIPS approved named algorithms:
 
 =item Digests
 
-SHA1, SHA224, SHA256, SHA384, SHA512, SHA512-224, SHA512-256,
```

Comment:
Specifying all the algorithms a provider supports duplicates the information so it potentially presents a maintenance problem. Also it conflicts with #10088 unless the manual page would be generated from a template. That would be one option. Another option would be to just reference an openssl provider -list-algos command or such when we will have it.

Reply:
Errrrrrr...  I dunno how it conflicts with #10088.  It's not like I'm making any implementation more or less available.  However, it might be good to note that the availability is subject to configuration and that for a definitive list relevant to the local installation, something like `openssl list -digest-algorithms` et al should be used.

Reply:
I kind of agree having just spent the time to update this list, that this is going to become a bit of a pain, so I'll replace it with a reference to `openssl list`. However I don't think that can do the asym algorithms yet (partly I guess because we are only part way through converting the asym stuff to providers).

Reply:
Yup...  I've code somewhere that expands what `openssl list` as well as `openssl provider`can display.

Reply:
Several manpages list algorithms -- and TLS versions.  Which ones are actually possible depends on the configuration, and a recent PR added the "ifdef" markup so that find-doc-nits wouldn't complain. Most of the pages, and the online help, say "any supported cipher" and the docs reference the openssl-cipher page.

I think having some typical options is worthwhile because it helps new users understand things better. The difference between AES, AES-GCM, AES-keywrap is not obvious, for example.

TL;DR (which should come at the top): you'll never satisfy all constraints.